client/src/components/Grid/configs/adminGroups.ts

```diff
@@ -0,0 +1,176 @@
+import { faEdit, faKey, faPlus, faTrash, faTrashRestore } from "@fortawesome/free-solid-svg-icons";
+import { useEventBus } from "@vueuse/core";
+import axios from "axios";
+
+import { deleteGroup, purgeGroup, undeleteGroup } from "@/api/groups";
+import Filtering, { contains, equals, toBool, type ValidFilter } from "@/utils/filtering";
+import { withPrefix } from "@/utils/redirect";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import type { ActionArray, FieldArray, GridConfig } from "./types";
+
+const { emit } = useEventBus<string>("grid-router-push");
+
+/**
+ * Local types
+ */
+type GroupEntry = Record<string, unknown>;
+
+/**
+ * Request and return data from server
+ */
+async function getData(offset: number, limit: number, search: string, sort_by: string, sort_desc: boolean) {
+    const query = {
+        limit: String(limit),
+        offset: String(offset),
+        search: search,
+        sort_by: sort_by,
+        sort_desc: String(sort_desc),
+    };
+    const queryString = new URLSearchParams(query).toString();
+    const { data } = await axios.get(withPrefix(`/admin/groups_list?${queryString}`));
+    return [data.rows, data.rows_total];
+}
+
+/**
+ * Actions are grid-wide operations
+ */
+const actions: ActionArray = [
+    {
+        title: "Create New Group",
+        icon: faPlus,
+        handler: () => {
+            emit("/admin/form/create_group");
+        },
+    },
+];
+
+/**
+ * Declare columns to be displayed
+ */
+const fields: FieldArray = [
+    {
+        key: "name",
+        title: "Name",
+        type: "operations",
+        operations: [
+            {
+                title: "Edit Name",
+                icon: faEdit,
+                condition: (data: GroupEntry) => !data.deleted,
+                handler: (data: GroupEntry) => {
+                    emit(`/admin/form/rename_group?id=${data.id}`);
+                },
+            },
+            {
+                title: "Edit Permissions",
+                icon: faKey,
+                condition: (data: GroupEntry) => !data.deleted,
+                handler: (data: GroupEntry) => {
+                    emit(`/admin/form/manage_users_and_roles_for_group?id=${data.id}`);
+                },
+            },
+            {
+                title: "Delete",
+                icon: faTrash,
+                condition: (data: GroupEntry) => !data.deleted,
+                handler: async (data: GroupEntry) => {
+                    try {
+                        await deleteGroup({ group_id: String(data.id) });
```

Comment:
I noticed that this action will delete the group right away. Should we display a confirmation message just is case of fat-fingering delete?
I missed it before, but the same happens with the other Roles grid too.

Reply:
Yes, definitely. I added a simple confirm alert for delete/purge operations in groups and roles. This should probably be replaced with a Vue-modal in the follow-ups.